src/ResourceManager/Common/Commands.Common.Strategies.UnitTest/TargetStateTest.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Microsoft.Rest;
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
+using System.Threading;
+using Xunit;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Commands.Common.Strategies.UnitTest
+{
+    public class TargetStateTest
+    {
+        class Client : ServiceClient<Client>, IClient
+        {
+            public T GetClient<T>() where T : ServiceClient<T>
+                => this as T;
+        }
+
+        class NestedModel
+        {
+            public string Name { get; set; }
+
+            public string Id { get; set; }
+        }
+
+        class Model
+        {
+            public string Location { get; set; }
+
+            public IList<NestedModel> Nested { get; set; }
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void Test()
```

Comment:
This should have a more descriptive method name - reflecting what this is testing (target state defaults?)